kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -382,8 +382,10 @@ private Field unionizeFields(Field firstField, Field secondField) {
       }
     }
 
-    checkState(firstField.getName().equals(secondField.getName()));
-    checkState(firstField.getType() == secondField.getType());
+    checkState(firstField.getName().equals(secondField.getName()),
+            "Cannot perform union operation on two fields having different names");
+    checkState(firstField.getType() == secondField.getType(),
+            "Cannot perform union operation on two fields having different datatypes");
```

Comment:
Uber suggestion @ManasjyotiSharma not inclined either way: Should we add the field name and types at the end of errorMessage? We can base our classification on the static part, but the customers on-prem as well will be able to neatly see what fields are we referring to here.

Reply:
Good suggestion @mukkachaitanya, let me include that!